### DISCUSSION TITLE
If I want to make my own event handler, where and how do I insert it?

### DISCUSSION THREAD

**Discussion by Marxsal** [TiddlyWiki/TiddlyWiki5#5490] [Q&A]
If I want to make my own event handler, where and how do I insert it?

Secondary question:  If I add a javascript tiddler, what determines when it executes? How can I make sure that it executes after other key objects (e.g. $tw.rootWidget) are initialized ?

Thanks!

**Comment by joshuafontany** [comment]
This will be a helpful set of Docs (most of the internals documentation are in the "/dev" edition):

https://tiddlywiki.com/dev/#Event%20Mechanism:%5B%5BEvent%20Mechanism%5D%5D%20%5B%5Bwikimethod%20module%20type%5D%5D%20BootMechanism%20StartupMechanism%20%5B%5BStartup%20Process%5D%5D%20HelloThere%20Introduction

**Reply by Marxsal** [reply]
Thanks Joshua. That was helpful for some of what we wanted. We're trying to rework Sini-Kit's code. SK has js script that is invoked from   

$:/core/templates/tiddlywiki5.html  

Obviously this would require writing over a core tiddler. To avoid this, for the moment, I've put the code in tiddler with a $:/raw... tag. This works fine, but does require an extra save and reload on the first use.

I thought of moving the code into a startup module per the documentation you pointed out, but that code appears to be sandboxed so you can't specify global functions nor even attach them to $tw . Any suggestions how to get the same effect as invoking at $:/core/templates/tiddlywiki5.html without actually hacking that core tiddler?

Thanks!

Edit: Thought of putting all the code into an event handler. But the code needs public callbacks from Google, so that doesn't work.
